workflows/testing/test_workflow/git.py

```diff
@@ -0,0 +1,20 @@
+import os
+import tempfile
+
+class GitRepository:
+    def __init__(self, url, ref):
+        self._url = url
+        self._dir = tempfile.TemporaryDirectory()
+        self.execute(f'git clone {url} {self._dir.name}')
```

Comment:
Do we need to worry about non-public repositories needing alternative credentials?

Reply:
I'm happy to say we don't support non-public repositories.